run.go

```diff
@@ -329,6 +329,11 @@ type RunVariable struct {
 	Value string `jsonapi:"attr,value"`
 }
 
+type RunVariableOption struct {
+	Key   string `json:"key"`
+	Value string `json:"value"`
+}
+
```

Comment:
Would it bet better to just redefine the RunVariable struct to be decorated as json: instead of jsonapi: ? I think that was the intent and it just got decorated incorrectly. By me, probably!

Reply:
I initially tried redefining the RunVariable, in that case it will marshal correctly from Go value to json but un-marshalling from json to Go value stops working. The only way I found to make both direction work is if I create the two distinct types.

Reply:
Yeah, I see what you mean, now. Since RunVariable was used as a create option, which more often times requires callers to specify the exact type name ([example](https://github.com/hashicorp/terraform/blob/76d40d281d5565d805017aeba4750191f063ab12/internal/cloud/backend_plan.go#L238-L247)), I think it would be better to redefine that as "json:" and create a new type for use in the Run struct. Like [this](https://github.com/hashicorp/go-tfe/compare/brandonc/reverse_type_fix) WDYT?

I think then we could get away with not reving the major version... Although we should still note this type change in the release notes in the context of a bug fix.

Reply:
Yes, this makes sense. I have renamed attributes to `RunVariable` and `RunVariableAttr`. This way all existing `Create` run calls will still work.